Makefile

```diff
@@ -1,51 +1,61 @@
-BUILD_ARGS ?= --summary all --verbose
-ZIG ?= zig
+BUILD_ARGS ?= --summary all --verbose -Dbootloader=$(BOOTLOADER)
+
+# get the last bootloader option
+BOOTLOADER ?= $(shell echo $$(ls ".bootloader-"* 2>/dev/null || echo grub) | cut -d'-' -f2)
```

Comment:
shouldn't the BOOTLOADER var be defined before being used for BUILD_ARGS?

Reply:
I don't think the order is really important here.

https://www.gnu.org/software/make/manual/html_node/Reading-Makefiles.html

> GNU make does its work in two distinct phases. During the first phase it reads all the makefiles, included makefiles, etc. and internalizes all the variables and their values and implicit and explicit rules, and builds a dependency graph of all the targets and their prerequisites. During the second phase, make uses this internalized data to determine which targets need to be updated and run the recipes necessary to update them

As you can read in the mentioned page, the '?=' operator is deferred to the second phase.

But if you want, I can move it to `BOOTLOADER` for greater clarity.

Reply:
I made the changes